services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/ApplicationConfig.java

```diff
@@ -286,19 +291,87 @@ public ObjectFactoryCreatingFactoryBean amqpServerFactory() {
         return factory;
     }
 
+    /**
+     * Exposes a factory for creating clients for the <em>AMQP Messaging Network</em> as a Spring bean.
+     * <p>
+     * The factory is initialized with the connection provided by {@link #downstreamConnection(Vertx)}.
+     *
+     * @param vertx The vert.x instance to run on.
+     *
+     * @return The factory.
+     */
+    @Bean
+    @Scope("prototype")
+    public DownstreamSenderFactory downstreamSenderFactory(final Vertx vertx) {
```

Comment:
Do you mind if I create an issue and do this after this PR, the JDBC PR and especially the integration tests are merged, so that I can test the migration more easily?

As this involves changing the AutoProvisioner this will not only affect the MongoDb registry but also the JDBC and File registry and would involve semi-manual re-testing on my machine for all of them.

Reply:
sure

Reply:
Created #2393